### PR TITLE
Put source under src folder, add test

### PR DESCRIPTION
**Pull request by rchristie**

### CODE REVIEW

File: tests/test_heart.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+from scaffoldmaker.utils.zinc_utils import evaluateFieldRange
+
+def assertAlmostEqualList(testcase, actualList, expectedList, delta):
+    assert len(actualList) == len(expectedList)
+    for actual, expected in zip(actualList, expectedList):
+        testcase.assertAlmostEqual(actual, expected, delta=delta)
+
+class HeartScaffoldTestCase(unittest.TestCase):
+
+    def test1_heart1(self):
```
Comment (hsorby) [inline tests/test_heart.py]: Test names should really start with `test_` this is what the test frameworks expect.

File: src/scaffoldmaker/utils/zinc_utils.py [hunk 2]
```diff
@@ -4,10 +4,34 @@
...
+
+    def __enter__(self):
+        self._object.beginChange()
+        return self
+
+    def __exit__(self, *args):
+        self._object.endChange()
+
+
 def getOrCreateCoordinateField(fieldmodule, name='coordinates', componentsCount=3):
```
Comment (hsorby) [inline src/scaffoldmaker/utils/zinc_utils.py]: These `getOrCreate*` functions could be canonicalised somewhat.

File: src/scaffoldmaker/utils/zinc_utils.py [hunk 3]
```diff
@@ -269,6 +286,24 @@ def getMaximumElementIdentifier(mesh):
...
+        fieldcache = fieldmodule.createFieldcache()
+        result, minimums = fieldNodesetMinimum.evaluateReal(fieldcache, componentsCount)
+        assert result == RESULT_OK
+        result, maximums = fieldNodesetMaximum.evaluateReal(fieldcache, componentsCount)
+        assert result == RESULT_OK
+        del fieldNodesetMinimum
+        del fieldNodesetMaximum
+    return minimums, maximums
+
 def interpolateNodesCubicHermite(cache, coordinates, xi, normal_scale, \
```
Comment (hsorby) [inline src/scaffoldmaker/utils/zinc_utils.py]: Functions like this utility should be in the `opencmiss.zinc.utils` package.

File: src/scaffoldmaker/utils/zinc_utils.py [hunk 4]
```diff
@@ -354,33 +389,32 @@ def exnodeStringFromNodeValues(
...
+            for v in range(nodeValueLabelsCount):
+                coordinates.setNodeParameters(cache, -1, nodeValueLabels[v], 1, nodeValues[n][v])
+        # serialise to string
+        sir = region.createStreaminformationRegion()
+        srm = sir.createStreamresourceMemory()
+        sir.setResourceGroupName(srm, groupName)
+        region.write(sir)
+        result, exString = srm.getBuffer()
+        #print('\n',nodeValues)
+        #print('exnodeStringFromNodeValues', result, exString)
```
Comment (hsorby) [inline src/scaffoldmaker/utils/zinc_utils.py]: Commented out debugging code should just be removed.

File: setup.py [hunk 5]
```diff
@@ -40,7 +40,7 @@ def readfile(filename, split=False):
     url='',
     license='APACHE',
     packages=find_packages(exclude=['ez_setup',]),
```
Comment (hsorby) [inline setup.py]: This line should be:
`packages=find_packages('src'),`

File: tests/test_heart.py (hunk 1 above)
Comment (rchristie) [reply tests/test_heart.py]: Will fix.

File: src/scaffoldmaker/utils/zinc_utils.py (hunk 2 above)
Comment (rchristie) [reply src/scaffoldmaker/utils/zinc_utils.py]: I'm not sure what you mean here?

File: setup.py (hunk 5 above)
Comment (rchristie) [reply setup.py]: Will fix.

Comment (rchristie) [reply setup.py]: You don't have it in mfxml2ex... what's the best usage here?
https://github.com/hsorby/mbfxml2ex/blob/master/setup.py uses:
`py_modules=["mbfxml2ex"],`
`package_dir={'': 'src'},`
Scaffoldmaker would have:
`packages=find_packages('src'),`
`package_dir={'': 'src'},`

Comment (hsorby) [reply setup.py]: No mbfxml2ex doesn't have any packages so it really doesn't need to use the `find_packages` helper function.

File: src/scaffoldmaker/utils/zinc_utils.py (hunk 2 above)
Comment (hsorby) [reply src/scaffoldmaker/utils/zinc_utils.py]: These functions are all pretty much the same there is a lot of code that could be reused.

File: src/scaffoldmaker/utils/zinc_utils.py (hunk 4 above)
Comment (rchristie) [reply src/scaffoldmaker/utils/zinc_utils.py]: Will fix.

**Pull request merged**